src/serde/mod.rs

```diff
@@ -0,0 +1,185 @@
+//! # Serialization and deserialization of Ion data
+//!
+//! This module offers APIs for serialization of Rust data structures into Ion data and deserialization of Ion data into Rust data structures.
+//! The APIs uses serde framework for serialization and deserialization. See the Serde website <https://serde.rs/> for additional documentation and usage examples.
+//! This feature doesn't support [Ion annotations] for serialization and deserialization.
```

Comment:
How are enum variants handled—or is that not supported yet? E.g.:
```rust
#[derive(Serialize, Deserialize)]
enum IpAddr {
    V4(String),
    V6(String),
}
```

What about tuple structs? Does this work? Is the serialized representation a s-expression?
```rust
#[derive(Serialize, Deserialize)]
struct Point(i32, i32);
```